src/extra/gfx/material_utils.hpp

```diff
@@ -76,15 +96,46 @@ inline void initConstantShaderParams(MaterialParameters &out, SHTable &paramsTab
   }
 }
 
-inline void initReferencedShaderParams(SHContext *shContext, SHShaderParameters &shShaderParameters, SHTable &paramsTable) {
+inline void initConstantTextureParams(MaterialParameters &out, SHTable &texturesTable) {

```

Comment:
I would have the *out* parameter always last (or always first) if possible for consistency across multiple functions.

In addition, shouldn't `texturesTable` be ` const &`? But maybe that's not good in C++ (I think it can impact perf?).

Reply:
Good catch, should be const yes (It's just a compiler annotation, no perf impact as far as I know)

Reply:
Moved out params to last